web/src/instrumentedComponents.tsx

```diff
@@ -63,7 +63,7 @@ export function InstrumentedTextField(
           ...(analyticsTags ?? {}),
         })
       }, textFieldEditDebounceMilliseconds),
-    [analyticsName, analyticsTags]
+    []
```

Comment:
hmmm...i think this will cause other problems? now multiple fields will share a single debouncer, which is not what you want.

Reply:
let's make the debouncer a global, and pass in all the arguments (rather than using the lambda to capture arguments from the outer scope)

that should make the state much easier to reason about, rather than having to think through how the debouncer state interacts with react state management

Reply:
the debouncing function is local to every instance of the `InstrumentedTextField` component, so we shouldn't hit any issues where intervals are reset by other fields' interactions, but i can refactor to make this code a little cleaner.